executor/src/witgen/global_constraints.rs

```diff
@@ -28,12 +29,17 @@ pub struct SimpleRangeConstraintSet<'a, T: FieldElement> {
     range_constraints: &'a BTreeMap<PolyID, RangeConstraint<T>>,
 }
 
-impl<'a, T: FieldElement> RangeConstraintSet<&AlgebraicReference, T>
+impl<'a, T: FieldElement> RangeConstraintSet<AlgebraicVariable<'a>, T>
     for SimpleRangeConstraintSet<'a, T>
 {
-    fn range_constraint(&self, id: &AlgebraicReference) -> Option<RangeConstraint<T>> {
-        assert!(!id.next);
-        self.range_constraints.get(&id.poly_id).cloned()
+    fn range_constraint(&self, id: AlgebraicVariable<'a>) -> Option<RangeConstraint<T>> {
+        match id {
+            AlgebraicVariable::Column(id) => {
+                assert!(!id.next);
+                self.range_constraints.get(&id.poly_id).cloned()
+            }
+            AlgebraicVariable::Public(_) => todo!(),
```

Comment:
```suggestion
            AlgebraicVariable::Public(_) => unimplemented!(),
```

Reply:
Will we have range constraints for publics?

Reply:
I'd say in theory yes; in practice, it doesn't make too much sense because the verifier might as well validate the range constraints themselves.